cmd/kube-controller-manager/app/options/options.go

```diff
@@ -49,32 +49,37 @@ import (
 	"github.com/spf13/pflag"
 )
 
+const (
+	// KubeControllerManagerUserAgent is the userAgent name when starting kube-controller managers.
+	KubeControllerManagerUserAgent = "kube-controller-manager"
+)
+
 // KubeControllerManagerOptions is the main context object for the kube-controller manager.
 type KubeControllerManagerOptions struct {
-	CloudProvider    *cmoptions.CloudProviderOptions
-	Debugging        *cmoptions.DebuggingOptions
-	GenericComponent *cmoptions.GenericComponentConfigOptions
-	KubeCloudShared  *cmoptions.KubeCloudSharedOptions
-
-	AttachDetachController           *cmoptions.AttachDetachControllerOptions
-	CSRSigningController             *cmoptions.CSRSigningControllerOptions
-	DaemonSetController              *cmoptions.DaemonSetControllerOptions
-	DeploymentController             *cmoptions.DeploymentControllerOptions
-	DeprecatedFlags                  *cmoptions.DeprecatedControllerOptions
-	EndPointController               *cmoptions.EndPointControllerOptions
-	GarbageCollectorController       *cmoptions.GarbageCollectorControllerOptions
-	HPAController                    *cmoptions.HPAControllerOptions
-	JobController                    *cmoptions.JobControllerOptions
-	NamespaceController              *cmoptions.NamespaceControllerOptions
-	NodeIpamController               *cmoptions.NodeIpamControllerOptions
-	NodeLifecycleController          *cmoptions.NodeLifecycleControllerOptions
-	PersistentVolumeBinderController *cmoptions.PersistentVolumeBinderControllerOptions
-	PodGCController                  *cmoptions.PodGCControllerOptions
-	ReplicaSetController             *cmoptions.ReplicaSetControllerOptions
-	ReplicationController            *cmoptions.ReplicationControllerOptions
-	ResourceQuotaController          *cmoptions.ResourceQuotaControllerOptions
-	SAController                     *cmoptions.SAControllerOptions
-	ServiceController                *cmoptions.ServiceControllerOptions
+	CloudProvider     *cmoptions.CloudProviderOptions
+	Debugging         *cmoptions.DebuggingOptions
+	GenericComponent  *cmoptions.GenericComponentConfigOptions
+	KubeCloudShared   *cmoptions.KubeCloudSharedOptions
+	ServiceController *cmoptions.ServiceControllerOptions
```

Comment:
@luxas out of curiosity: is this disabled somehow when an external cloud ctrl manager is running?

Reply:
cc @andrewsykim 
@sttts IIRC, yes it is
@sttts might be interesting reading: https://docs.google.com/document/d/1YqsPzeG10E59vea5StKFM6W_xZq034eTaf1n649JQvs/edit?ouid=100233907167775907499&usp=docs_home

Reply:
yes, service controller does nothing if cloud provider is not set or set to `external`

Reply:
If you're curious https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/service/service_controller.go#L225-L227

Reply:
Also: https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-controller-manager/app/controllermanager.go#L345-L350